allennlp/predictors/dialog_qa.py

```diff
@@ -0,0 +1,56 @@
+import json
+from overrides import overrides
+
+from allennlp.common.util import JsonDict
+from allennlp.data import DatasetReader, Instance
+from allennlp.data.tokenizers.word_splitter import SpacyWordSplitter
+from allennlp.predictors.predictor import Predictor
+from allennlp.models import Model
+
+
+@Predictor.register('dialog_qa')
+class DialogQAPredictor(Predictor):
+    def __init__(self, model: Model, dataset_reader: DatasetReader) -> None:
+        super().__init__(model, dataset_reader)
+        self._tokenizer = SpacyWordSplitter(language='en_core_web_sm')
+
+    def predict(self, jsonline: str) -> JsonDict:
+        """
+        Make a machine comprehension prediction on the supplied input.
+       -------
+        A dictionary that represents the prediction made by the system.  The answer string will be under the
+        "best_span_str" key.
+        """
+        return self.predict_json(json.loads(jsonline))
+
+    @overrides
+    def _json_to_instance(self, json_dict: JsonDict) -> Instance:
+        """
+        Expects json that looks like the original quac file
+        """
+        paragraph_json = json_dict["paragraphs"][0]
+        paragraph = paragraph_json['context']
+        tokenized_paragraph = self._tokenizer.split_words(paragraph)
+        qas = paragraph_json['qas']
+        metadata = {}
+        metadata["instance_id"] = [qa['id'] for qa in qas]
```

Comment:
Because the main purpose of the predictor is to run a demo, you should probably be robust to not having an id here.

Reply:
i can remove the instance id, but please refer to the comments to the other feedback (a.k.a. why i can't remove answers).